x/mongo/driver/operation.go

```diff
@@ -605,6 +611,12 @@ func (op Operation) roundTrip(ctx context.Context, conn Connection, wm []byte) (
 		return nil, Error{Message: err.Error(), Labels: labels, Wrapped: err}
 	}
 
+	// If we're using a streamable connection, we set its streaming state based on the moreToCome flag in the server
+	// response.
+	if streamer, ok := conn.(StreamerConnection); ok {
+		streamer.SetStreaming(wiremessage.IsMsgMoreToCome(wm))
```

Comment:
What if we aren't? What if the server sends moreToCome and our conn is _not_ a streaming connection? Should this be an error? How does the rest of the code handle this?

Reply:
Right now, we'd effectively ignore the `moreToCome` flag in the response if the connection isn't a `StreamerConnection`. This was already the behavior before this PR, though, because we were never even checking this flag in the response.